Assets/Scripts/Door.cs

```diff
@@ -21,12 +26,31 @@ public class Door : InteractiveObject
     [SerializeField]
     private AudioClip openAudioClip;
 
-    public override string DisplayText => isLocked ? lockedDisplayText :  base.DisplayText;
+    [SerializeField]
+    private bool willLoadNextScene = false;
+
+    //public override string DisplayText => isLocked ? lockedDisplayText :  base.DisplayText;
+
+    public override string DisplayText
+    {
+        get
+        {
+            string toReturn;
+
+            if (isLocked)
+                toReturn = HasKey ? $"Use {key.ObjectName}" : lockedDisplayText;
+            else
+                toReturn = base.DisplayText;
+
+            return toReturn;
+        }
+    }
 
-    
 
+    private bool HasKey => PlayerInventory.InventoryObjects.Contains(key);
```

Comment:
You should put all of these variables right beneath the SerializeFields, and put them into a region.